test/core/symbol/MerkleHashBuilder.spec.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Converter, MerkleHashBuilder } from '@core';
+import { expect } from 'chai';
+import * as Crypto from 'crypto';
+
+describe('MerkleHashBuilder should', () => {
+    const calculateMerkleHash = (hashes: Uint8Array[]) => {
+        const builder = new MerkleHashBuilder();
+        hashes.forEach((hash) => builder.update(hash));
+        return builder.final();
+    };
+
+    it('Can build from zero hashes', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder();
+
+        // Act:
+        const rootHash = builder.final();
+
+        // Assert:
+        expect(rootHash).deep.equal(new Uint8Array(32));
+    });
+
+    it('Can build from one child', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder().update(
+            Converter.hexToUint8('215B158F0BD416B596271BCE527CD9DC8E4A639CC271D896F9156AF6F441EEB9'),
+        );
+
+        // Act:
+        const rootHash = builder.final();
+
+        // Assert:
+        expect(Converter.uint8ToHex(rootHash)).equal('215B158F0BD416B596271BCE527CD9DC8E4A639CC271D896F9156AF6F441EEB9');
+    });
+
+    it('Can build from blanced tree', () => {
+        // Arrange:
+        const builder = new MerkleHashBuilder()
+            .update(Converter.hexToUint8('215b158f0bd416b596271bce527cd9dc8e4a639cc271d896f9156af6f441eeb9'))
```

Comment:
generally would try to be consistent and use upper (or lower) hex characters everywhere instead of having some strings upper and some lower